### PR TITLE
Formatting

### LINKED ISSUE #115: format support automagic to_string
following should work

``` pan
a = list(1, 2);
format("woohoo %s", a);
```

where any formatted resources should be converted to strings using `to_string`

### PR DESCRIPTION
**Pull request by iliclaey**
Fixes #115 and #116

### CODE REVIEW

File: panc-docs/source/standard-functions/standard-functions.rst [hunk 1]
```diff
@@ -570,16 +576,17 @@ format -- format a string by replacing references to parameters
...
 -----------
 
 The ``format`` function will replace all references within the ``fmt``
-string with the values of the referenced properties. This provides
+string with the values of the referenced elements. This provides
 functionality similar to the c-language's ``printf`` function. The
 syntax of the ``fmt`` string follows that provided in the java language;
-see the Formatter entry for full details.
+see the Formatter entry for full details. When passing a resource as an
+argument, the string replacement field should be used.
```
Comment (stdweird) [inline panc-docs/source/standard-functions/standard-functions.rst]: what is this about?

File: panc/src/test/pan/Functionality/format/format10.pan [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+object template format10;
+
+variable HASH = dict("entry1", 1, "entry2", 2);
+variable STR = format("%s", HASH);
+
+'/result' = {
+    expected = "{ entry1, 1, entry2, 2 }";
+    STR == expected;
+};
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/format/format10.pan]: add EOF

File: panc/src/test/pan/Functionality/format/format12.pan [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#
+
+object template format12;
+
+variable STR = format("%.2f", 1.15);
+
+'/result' = {
+    expected = "1.15";
+    STR == expected;
+};
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/format/format12.pan]: newline EOF

File: panc/src/test/pan/Functionality/function/function16.pan [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# error() also allows formatting
+# @expect=org.quattor.pan.exceptions.EvaluationException ".*user-initiated error:.*"
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/function/function16.pan]: also check the the message has the correct formatting (if possible)

File: panc/src/test/pan/Functionality/function/function17.pan [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# error() formatting works with resources
+# @expect=org.quattor.pan.exceptions.EvaluationException ".*user-initiated error:.*"
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/function/function17.pan]: check formatting in message

File: panc/src/test/pan/Functionality/function/function18.pan [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#
+# @expect=org.quattor.pan.exceptions.EvaluationException ".*user-initiated error thrown with non-string argument.*"
```
Comment (stdweird) [inline panc/src/test/pan/Functionality/function/function18.pan]: might be better if this was a plain syntax error

File: panc-docs/source/standard-functions/standard-functions.rst (hunk 1 above)
Comment (iliclaey) [inline panc-docs/source/standard-functions/standard-functions.rst]: I meant to say you need to use "%s" when passing a resource

File: panc/src/main/java/org/quattor/pan/dml/functions/Debug.java [hunk 7]
```diff
@@ -47,17 +48,17 @@ private Debug(SourceRange sourceRange, Operation... operations)
...
 		}
-		if (operations[0] instanceof Element) {
-			if (!(operations[0] instanceof StringProperty)) {
-				throw SyntaxException.create(sourceRange,
-						MSG_ONE_STRING_ARG_REQ, "debug");
+
+		// Ensure that if there is exactly one argument, it is a string.
+		if (operations.length == 1) {
+			if (operations[0] instanceof Element) {
+				if (!(operations[0] instanceof StringProperty)) {
```
Comment (piojo) [inline panc/src/main/java/org/quattor/pan/dml/functions/Debug.java]: Please use multiple conditions together:

```java
if (operations.length == 1
    && operations[0] instanceof Element
    && !(operations[0] instanceof StringProperty)) {
    throw ...
}

**Pull request merged**